src/redisearch_rs/trie_bencher/src/bencher.rs

```diff
@@ -162,6 +162,42 @@ impl OperationBencher {
         load_c_benchmark(&mut group, &self.keys);
         group.finish();
     }
+
+    /// Benchmark the find prefixes iterator.
+    ///
+    /// The benchmark group will be marked with the given label.
+    pub fn find_prefixes_group(&self, c: &mut Criterion, target: &str, label: &str) {
+        let mut group = self.benchmark_group_immutable(c, label);
+        find_prefixes_rust_benchmark(&mut group, &self.rust_map, target);
+        find_prefixes_c_benchmark(&mut group, &self.keys, target);
+        group.finish();
+    }
+}
+
+fn find_prefixes_rust_benchmark<M: Measurement>(
+    c: &mut BenchmarkGroup<'_, M>,
+    map: &RustTrieMap,
+    target: &str,
+) {
+    let target = target.as_bytes();
+    c.bench_function("Rust", |b| {
+        // Set to avoid timing the drop phase to match the performance profile of the C bench.
+        b.iter_with_large_drop(|| map.prefixes_iter(black_box(target)).collect::<Vec<_>>())
+    });
+}
+
+fn find_prefixes_c_benchmark<M: Measurement>(
+    c: &mut BenchmarkGroup<'_, M>,
+    terms: &[String],
+    target: &str,
+) {
+    let target = target.into_cstring();
+    let view = target.as_view();
+    let map = c_load_from_terms(terms);
+    c.bench_function("C", |b| {
+        // We are leaking memory here, since we don't free the pointer to the result array.
+        b.iter_with_large_drop(|| map.find_prefixes(view).unwrap())
```

Comment:
This can be fixed by moving `util/arr.*` to a subfolder, adding a `CMakeLists.txt` file, build them as a standalone static library, binding them to Rust and then calling `array_free` here.

It feels like a lot of changes for benching code that will be deleted once we swap the impl, but open to making them if you think it's worthwhile. cc @raz-mon @joanFM

Reply:
Done in 9517ea36f

Reply:
Found a simpler way in e5cfc7a58 to avoid touching as many C files.